reya-rust-sdk/src/http_provider.rs

```diff
@@ -213,6 +207,89 @@ impl HttpProvider {
         eyre::Ok(receipt.transaction_hash)
     }
 
+    ///
+    /// execute a batch of orders
+    ///
+    pub async fn execute_batch(
+        &self,
+        signer: PrivateKeySigner,
+        batch_orders: &mut Vec<data_types::BatchOrder>,
+    ) -> eyre::Result<B256> // return the transaction hash
+    {
+        //
+        let mut orders: Vec<CoreProxy::ConditionalOrderDetails> = vec![];
+        let mut signatures: Vec<CoreProxy::EIP712Signature> = vec![];
+
+        // create http provider
+        let provider = ProviderBuilder::new()
+            .with_recommended_fillers()
+            .wallet(EthereumWallet::from(signer))
+            .on_http(self.sdk_config.rpc_sdk_url.clone());
+
+        let core_proxy = CoreProxy::new(
+            self.sdk_config.order_gateway_contract_address.parse()?,
+            provider.clone(),
+        );
+
+        // add all order ans signatures to the batch vector
+        for batch_order in batch_orders {
+            //trace!("Batch execute orders {}", batch_order);
+
+            let mut encoded_inputs: Vec<u8> = Vec::new();
+            if batch_order.order_type == data_types::OrderType::StopLoss {
+                // generate encoded core command for the input bytes of a stop_loss order
+                // The input byte structure is:
+                // {
+                //     trigger_price, // stop_price!
+                //     price_limit,   // price limit is the slippage tolerance,we can set it to max uint or zero for now depending on the direction of the trade
```

Comment:
"we can set it to max uint or zero for now depending on the direction of the trade" should re-open this convo on discor